### ISSUE TITLE
`getip.sh` isn't reliable enough

### ISSUE THREAD

**Issue by jakebiesinger** [uci-cbcl/genomix#31]
@JavierJia @Nan-Zhang 

We've seen from #29 and #26 that java returns a different value than `getip.sh`.  `getip.sh` is just a hack that reads out the IP address from `ifconfig` for `eth0`.

In other words, it's a hack.  Users may not have their setup on `eth0`-- I imagine that's the error you saw, @JavierJia ? your laptop is probably connecting over wifi and therefore doesn't use eth0? We're seeing this on travis, where some machines [aren't connected on eth0](https://travis-ci.org/uci-cbcl/genomix/jobs/12915715) and others are...

I think the solution is probably a very small java program that uses the java API rather than parsing from eth0.

Any volunteers to implement such a program?

**Comment by JavierJia** [comment]
eth0 is not necessarily a wired port. My machine also have eth0. Let me fix it. The previous Java solution only not work because it return an 127.0.0.1, which not recognized by Hyracks :cry:  . Let me fix it.

**Comment by jakebiesinger** [comment]
But it's not guaranteed that it will be there, as I pointed out in the travis builds. For example, https://travis-ci.org/uci-cbcl/genomix/jobs/12915715#L1678

**Comment by jakebiesinger** [comment]
I thought the problem wasn't so much 127.0.0.1 as it was that the CC was set to the proper address, while the driver was trying to connect via 127.0.0.1.  I think it's the discrepancy, not the value that's wrong.

The `startCC.sh` and `startNC.sh` scripts all use `getip.sh`... Seems like we should try to be consistent about this...

**Comment by JavierJia** [comment]
well, that's actually need we to change getip.sh.

**Comment by jakebiesinger** [comment]
Yep.  and getip.sh is annoying 'cause it means you can't run anything without being inside of a mvn built package-- those scripts must be present and must be executable to even get your own IP address...

**Comment by JavierJia** [comment]
You are the python expert, do you think this line will sufficient enough? (using bash script will inevitably jump into endless hardware issues)

```
python -c 'import socket; print socket.gethostbyname(socket.getfqdn())'
```

**Comment by jakebiesinger** [comment]
hmm.. I guess we really don't want the 127.0.0.1 address at all.  The slaves can't connect to that address and expect to reach the CC.  googling around tells me that eth0 really is the place to look, at least in most cases where NAT isn't involved (which it is on the amazon server travis is running on).

The python script will also get the loopback interface, yes?

**Comment by jakebiesinger** [comment]
see http://www.if-not-true-then-false.com/2010/linux-get-ip-address/ for more examples, using either eth0 or some external website (which wouldn't work for most of our cases-- the nodes are generally not publicly available).
